test/test_unistd.rs

```diff
@@ -599,6 +600,63 @@ fn test_symlinkat() {
     );
 }
 
+
+#[test]
+fn test_unlinkat_dir_noremovedir() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let dirname = "foo_dir";
+    let dirpath = tempdir.path().join(dirname);
+
+    // Create dir
+    DirBuilder::new().recursive(true).create(&dirpath).unwrap();
+
+    // Get file descriptor for base directory
+    let dirfd = fcntl::open(tempdir.path(), fcntl::OFlag::empty(), stat::Mode::empty()).unwrap();
+
+    // Attempt unlink dir at relative path without proper flag
+    let err_result = unlinkat(Some(dirfd), dirname, UnlinkatFlags::NoRemoveDir).unwrap_err();
+    match err_result {
+        Error::Sys(Errno::EISDIR) => println!("{}", err_result),
```

Comment:
passing tests shouldn't print anything.  Just do `assert!(err_result == Error::Sys(Errno::EISDIR) || err_result == Error::Sys(Errno::EPERM))`

Reply:
I see .. good catch/comment. I've corrected this in the 5th commit.